tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -2921,6 +2997,30 @@ static void build_image(const char *output,
 		os_image_partition->size = kernel.st_size;
 	}
 
+	if (strcasecmp(info->id, "DECO-M5-EU-V3.0") == 0) {
```

Comment:
What's that supposed to do. This should be discussed in your commit message. If you have to introduce that much logic, I'd even consider to make the tplink-safeloader changes a separate commit before adding the actual device.